iati/core/schemas.py

```diff
@@ -52,25 +52,19 @@ def __init__(self, name=None):
             Create test instance where the SchemaError is raised.
 
         """
-        self.name = name
         self._schema_base_tree = None
         self.codelists = set()
         self.rulesets = set()
+        self.source_path = path
 
-        if isinstance(name, str):
-            path = iati.core.resources.get_schema_path(self.name)
-            try:
-                loaded_tree = iati.core.resources.load_as_tree(path)
-            except (IOError, OSError):
-                msg = "Failed to load tree at '{0}' when creating Schema.".format(path)
-                iati.core.utilities.log_error(msg)
-                raise iati.core.exceptions.SchemaError
-            else:
-                self._schema_base_tree = loaded_tree
-        elif name is not None:
-            msg = "The name of the Schema is an invalid type. Must be a string, though was a {0}.".format(type(name))
+        try:
+            loaded_tree = iati.core.resources.load_as_tree(path)
+        except (IOError, OSError):
```

Comment:
Is one of these Python2, the other 3? If so, could be highlighted in some manner - it'd be a new case, so am not entirely sure how to note this.